bindings/c/test/fdb_api.hpp

```diff
@@ -556,6 +559,58 @@ class Database {
 	}
 };
 
+class Tenant final {
+	static const std::string tenantManagementMapPrefix;
+	static const std::string tenantMapPrefix;
+
+public:
+	static TypedFuture<future_var::None> createTenant(Transaction tr, std::string name) {
+		tr.setOption(FDBTransactionOption::FDB_TR_OPTION_SPECIAL_KEY_SPACE_ENABLE_WRITES, 1);
+		tr.setOption(FDBTransactionOption::FDB_TR_OPTION_LOCK_AWARE, 1);
+		KeyRef tenantManagementKey = toBytesRef(tenantManagementMapPrefix + name);
+		tr.set(tenantManagementKey, toBytesRef(std::string("")));
+		return tr.commit();
+	}
+
+	static TypedFuture<future_var::None> deleteTenant(Transaction tr, std::string name) {
+		tr.setOption(FDBTransactionOption::FDB_TR_OPTION_RAW_ACCESS, 1);
+		tr.setOption(FDBTransactionOption::FDB_TR_OPTION_LOCK_AWARE, 1);
+		KeyRef tenantManagementKey = toBytesRef(tenantManagementMapPrefix + name);
+		tr.clear(tenantManagementKey);
+		return tr.commit();
+	}
+
+	Tenant(fdb::Database* db, const uint8_t* name, int name_length) {
```

Comment:
This should not be public. The idea is to have the usage of the fdb_c API fully encapsulated in these classes without ever having to take or pass the native handle. Exposing native handles to be used breaks the abstraction.

Reply:
I tried refactoring the code to match the Database class, but I still don't quite follow what you mean here. Can you explain further? This is the Tenant constructor which gives us a starting object to work with similar to the Database one. Is there something I exposed here that should be adjusted to avoid native handles? I thought `fdb::Database` would be preferred over `native::FDBDatabase`, but I am glad to make any adjustments as needed